### PR TITLE
ROX-11279: Disable removed default policies during migration to Postgres

### PR DESCRIPTION
**Pull request by c-du**
## Description
We store removed_default_policies in Rocksdb which would block default policies to be added to policies primary table. We do not allow default policies to be removed since 3.66.
While migrating to Postgres, we plan to remove this table and disable the default policies that were removed before. 
The default policies are a copy of current of default policies without the new default policies added during 3.66 to 3.73. 
The following is a list of default policies that are NOT included in this migration.
```
automount_service_account_token.json
deployment_has_ingress_network_policy.json
external_exposure.json
log4shell.json
privilege_escalation.json
severity_high_privileged.json
spring4shell.json
```
## Checklist
- [ ] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- [ ] Evaluated and added CHANGELOG entry if required
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

File: pkg/booleanpolicy/policyversion/convertor.go [hunk 1]
```diff
@@ -27,16 +33,20 @@ func EnsureConvertedToLatest(p *storage.Policy) error {
...
-	// If it's not the latest version, delegate to the upgrader
-	// CurrentVersion should always be the latest, thus this will always involve an upgrade.
-	if !IsCurrentVersion(ver) {
-		if err := upgradePolicyTo(p, CurrentVersion()); err != nil {
+	switch diff := Compare(ver, toVersion); {
+	case diff > 0:
+		// No downgrade
+		utils.CrashOnError(errors.Errorf("Unexpected version %s, cannot downgrade policy version to %s", ver.String(), toVersion.String()))
+	case diff < 0:
+		// If it's blow the requested version, delegate to the upgrader
```
Comment (dashrews78) [inline pkg/booleanpolicy/policyversion/convertor.go:41]: nit:  below

File: migrator/migrations/n_39_to_n_40_postgres_policies/legacy/store.go [hunk 2]
```diff
@@ -24,6 +28,7 @@ type Store interface {
 // New returns a new Store instance using the provided bolt DB instance.
 func New(db *bolt.DB) Store {
 	bolthelper.RegisterBucketOrPanic(db, policyBucket)
+	bolthelper.RegisterBucketOrPanic(db, removedDefaultPolicyBucket)
```
Comment (dashrews78) [inline migrator/migrations/n_39_to_n_40_postgres_policies/legacy/store.go:31]: Does this bucket still exist after 66?  For instance if a customer started at 70, do we have the bucket just empty?

Comment (c-du) [reply migrator/migrations/n_39_to_n_40_postgres_policies/legacy/store.go:31]: Yes, it still exists. If a customer started at 70, it is just empty but we cannot assume that.

**Review by dashrews78 — APPROVED** [review]
LGTM

File: migrator/migrations/n_39_to_n_40_postgres_policies/legacy/remove_test.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
+package legacy
+
+import (
+	"context"
+	"encoding/json"
```
Comment (connorgorman) [inline migrator/migrations/n_39_to_n_40_postgres_policies/legacy/remove_test.go:5]: i think this needs to be jsonpb

File: migrator/migrations/n_39_to_n_40_postgres_policies/legacy/store_impl.go [hunk 4]
```diff
@@ -5,20 +5,24 @@ package legacy
 
 import (
 	"context"
+	"encoding/json"
```
Comment (connorgorman) [inline migrator/migrations/n_39_to_n_40_postgres_policies/legacy/store_impl.go:8]: jsonpb I think

File: migrator/migrations/n_39_to_n_40_postgres_policies/legacy/remove_test.go (hunk 3 above)
Comment (c-du) [reply migrator/migrations/n_39_to_n_40_postgres_policies/legacy/remove_test.go:5]: This is only used to marshal a boolean, not a proto message.

File: migrator/migrations/n_39_to_n_40_postgres_policies/legacy/store_impl.go (hunk 4 above)
Comment (c-du) [reply migrator/migrations/n_39_to_n_40_postgres_policies/legacy/store_impl.go:8]: This is existing boolean in json.

**Pull request merged**